### PR TITLE
Add custom methods to the interfaces

### PR DESCRIPTION
**Pull request by jhedstrom**
This addresses #54.

I realized I need not wait for #39 to be merged.

### CODE REVIEW

File: src/MessageTypeInterface.php [hunk 1]
```diff
@@ -14,4 +14,140 @@
...
+   *   @todo Why does this return string or array?
+   */
+  public function getText($langcode = NULL, array $options = array());
+
+  /**
+   * Check if the message is new.
+   *
+   * @return bool
+   */
+  public function isLocked();
```
Comment (amitaibu) [inline src/MessageTypeInterface.php:154]: what is `isLocked`?

Comment (jhedstrom) [inline src/MessageTypeInterface.php:154]: Unsure--it appears to return the opposite of `isNew()`.

**Comment by jhedstrom** [comment]
I've also started adding unit tests for the custom entity methods.

File: src/MessageTypeInterface.php (hunk 1 above)
Comment (jhedstrom) [inline src/MessageTypeInterface.php:154]: It isn't used anywhere, so we could axe it.

File: src/MessageInterface.php [hunk 2]
```diff
@@ -8,10 +8,139 @@
...
+   *
+   * @return $this
+   */
+  public function setCreatedTime($timestamp);
+
+  /**
+   * Retrieve the message owner object.
+   *
+   * @return \Drupal\user\UserInterface
+   *  The user object.
```
Comment (RoySegall) [inline src/MessageInterface.php:56]: Missing extra space.

File: src/MessageInterface.php [hunk 3]
```diff
@@ -8,10 +8,139 @@
...
+   */
+  public function getAuthor();
+
+  /**
+   * Retrieve the author ID.
+   *
+   * @return int
+   *  The author ID.
```
Comment (RoySegall) [inline src/MessageInterface.php]: Missing extra space.

File: src/MessageInterface.php [hunk 4]
```diff
@@ -8,10 +8,139 @@
...
+  public function getAuthorId();
+
+  /**
+   * Set the author ID.s.
+   *
+   * @param int $uid
+   *   The user ID.
+   *
+   * @return $this
+   *  The message object.
```
Comment (RoySegall) [inline src/MessageInterface.php]: Missing extra space.

File: src/MessageInterface.php [hunk 5]
```diff
@@ -8,10 +8,139 @@
...
+   *
+   * @return string
+   */
+  public function getUUID();
+
+  /**
+   * Retrieve the message arguments.
+   *
+   * @return array
+   *  The arguments of the message.
```
Comment (RoySegall) [inline src/MessageInterface.php]: Missing extra space.

File: src/MessageInterface.php [hunk 6]
```diff
@@ -8,10 +8,139 @@
...
+   * Replace arguments with their placeholders.
+   *
+   * @param $langcode
+   *   Optional; The language to get the text in. If not set the current language
+   *   will be used.
+   * @param $options
+   *   Optional; Array to be passed to MessageType::getText().
+   *
+   * @return string
+   *  The message text.
```
Comment (RoySegall) [inline src/MessageInterface.php]: Missing extra space.

File: src/MessageTypeInterface.php (hunk 1 above)
Comment (RoySegall) [inline src/MessageTypeInterface.php:154]: Any way. I'll merge it and remove it.

Comment (RoySegall) [inline src/MessageTypeInterface.php:154]: No, sorry. I'll keep. It's used in `MessageTypeForm::form()`:

``` php
    $form['type'] = array(
      '#type' => 'machine_name',
      '#disabled' => $type->isLocked(),
    );
```

**Comment by RoySegall** [comment]
@jhedstrom There are some conflicts. Can you solve them? 10x.

**Comment by jhedstrom** [comment]
@RoySegall those issues should be fixed now.

**Comment by RoySegall** [comment]
10x for the changes but there still some merge conflicts. Have you merged the latest changes from `update-travis`?

**Comment by jhedstrom** [comment]
Oops, I forgot about those. I'll merge and push shortly.

**Comment by jhedstrom** [comment]
@RoySegall I rebased off of `update-travis`.

**Comment by mccrodp** [comment]
Great, looks like you got the changes from this old PR in too! https://github.com/Gizra/message/pull/50

**Comment by RoySegall** [comment]
10x for the changes. The tests are failing but I'll have a look on that one before the merging.

**Comment by RoySegall** [comment]
I can see why the tests are failing. You still have the `getAuthor...` methods when @chx remove them in one of his PRs. I'll handle it and merge it.

**Pull request merged**